### PR TITLE
Migrate to Kotlin DSL

### PR DESCRIPTION
**Pull request by onlymash**

### CODE REVIEW

**Comment by Mygod** [comment]
I've been meaning to implement this so this is great, but could you make build pass first?

**Comment by onlymash** [comment]
```
error computing cache key: template: 
cacheKey:1:8: executing "cacheKey" at <checksum "build.gradle">: 
error calling checksum: open /home/circleci/code/build.gradle: 
no such file or directory
```

This is not my fault

**Comment by onlymash** [comment]
Update the circleci config
Screenshot_20200412-082116

**Comment by Mygod** [comment]
Thanks!

File: plugin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+}
+
+dependencies {
+    val junitVersion = rootExt.get("junit_version") as String
+    val androidTestVersion = rootExt.get("android_test_version") as String
+    val androidEspressoVersion = rootExt.get("android_espresso_version") as String
+
+    api(kotlin("stdlib-jdk8", KotlinCompilerVersion.VERSION))
```
Comment (Mygod) [inline plugin/build.gradle.kts:54]: Shouldn't this use `kotlinVersion`?

Comment (onlymash) [reply plugin/build.gradle.kts:54]: KotlinCompilerVersion.VERSION same as kotlinVersion

**Comment by Mygod** [comment]
Could you merge master and resolve the conflicts?

Also please revert 0ef147b8583a0c3a5acac946f9e0839650a118d0, if any lint check fails I will take care of that for you.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+tasks.register("clean", Delete::class) {
+    delete(rootProject.buildDir)
+}
+
+tasks.withType<com.github.benmanes.gradle.versions.updates.DependencyUpdatesTask> {
+    checkForGradleUpdate = true
+    outputFormatter = "json"
+    outputDir = "build/dependencyUpdates"
+    reportfileName = "report"
+}
```
Comment (Mygod) [inline build.gradle.kts:55]: What is this for?

Comment (onlymash) [reply build.gradle.kts:55]: https://github.com/ben-manes/gradle-versions-plugin

Comment (Mygod) [reply build.gradle.kts:55]: All of these seem optional. I think the default behavior is reasonable.

File: plugin/build.gradle.kts (hunk 1 above)
Comment (Mygod) [reply plugin/build.gradle.kts:54]: I don't think so. It seems to be decided by gradle but I will handle it.

**Pull request merged**